torch/_higher_order_ops/flex_attention.py

```diff
@@ -137,17 +148,44 @@ def math_attention(
     m = torch.arange(0, scores.size(2), device=scores.device)
     n = torch.arange(0, scores.size(3), device=scores.device)
 
-    in_dim_buffers = (None,) * len(other_buffers)
-    score_mod = torch.vmap(score_mod, in_dims=(0, None, None, None, 0) + in_dim_buffers)
-    score_mod = torch.vmap(score_mod, in_dims=(0, None, None, 0, None) + in_dim_buffers)
-    score_mod = torch.vmap(score_mod, in_dims=(0, None, 0, None, None) + in_dim_buffers)
-    score_mod = torch.vmap(score_mod, in_dims=(0, 0, None, None, None) + in_dim_buffers)
+    score_mod_in_dim_buffers = (None,) * len(score_mod_other_buffers)
+    score_mod = torch.vmap(
```

Comment:
I feel like there's an utility function we could make here for all the use-cases of this pattern.

Reply:
refactor out a ```math_attention_inner``` as both fwd and bwd would use the whole fwd pass.

Reply:
I meant to refactor out the `vmap` pattern, not the `math_attention_inner`.